### ISSUE TITLE
error: syntax error at or near "bcb1b9c6c4f70e9890e1c5075984acec5ee02"

### ISSUE THREAD

**Issue by BerkeleyTrue** [loopbackio/loopback-connector-postgresql#34]
When creating a new model in explorer, this error is produced. Model has trackChanges: true. If this is removed then creating models works as expected.

Using
  loopback 2.4.0
  loopback-connector-postgresql 1.1.4

```
/Users/berkeley/node/loopback/gems/node_modules/loopback/lib/models/change.js:453
    throw err;
          ^
error: syntax error at or near "bcb1b9c6c4f70e9890e1c5075984acec5ee02"
    at Connection.parseE (/Users/berkeley/node/loopback/gems/node_modules/loopback-connector-postgresql/node_modules/pg.js/lib/connection.js:534:11)
    at Connection.parseMessage (/Users/berkeley/node/loopback/gems/node_modules/loopback-connector-postgresql/node_modules/pg.js/lib/connection.js:361:17)
    at Socket.<anonymous> (/Users/berkeley/node/loopback/gems/node_modules/loopback-connector-postgresql/node_modules/pg.js/lib/connection.js:105:22)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:764:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:426:10)
    at emitReadable (_stream_readable.js:422:5)
    at readableAddChunk (_stream_readable.js:165:9)
    at Socket.Readable.push (_stream_readable.js:127:10)
```

**Comment by dpkirchner** [comment]
I think this ticket should probably be re-opened. Quick summary: loopback-connector-postgresql can pass unquoted strings directly to a query built through concatenation, if the model expects a number for that column:

https://github.com/strongloop/loopback-connector-postgresql/blob/master/lib/postgresql.js#L491

More details that may help build a repro:

I ran in to this same issue. The problem, for me, came up when I switched databases from mongo to postgresql and then revisited my site without clearing cookies.

The specific error message ("syntax error at or near") actually comes directly from postgres, not node. However, it looks like the error is because some raw value (your `bcb*` string) is being passed to the database directly. A console.log in the postgresql query method, in my case, shows:

```
SELECT "realm", "username", "password", "credentials", "challenges",
"email", "emailverified", "verificationtoken", "status", "created",
"lastupdated", "id"  FROM "db"."user" WHERE
"id" = 54abd18d447071ad2c2be011  ORDER BY "id" LIMIT 1
```

Anyway, I don't know the exact set of steps to follow to reproduce this scenario (I was in the middle of something else). I just know that the id value is not being quoted.

Ultimately, the issue comes up because the model expects a number but is getting a string from redis, where my sessions are stored. It might not make sense to patch the library to reject non-numeric values but it seems prudent to replace the query builder with something that uses parameters.

FWIW, I'm using loopback-connector-postgresql 1.2.2 installed via npm.
